app/bundles/UserBundle/Security/Authenticator/Passport/Badge/PasswordStrengthBadge.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\UserBundle\Security\Authenticator\Passport\Badge;
+
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\BadgeInterface;
+
+class PasswordStrengthBadge implements BadgeInterface
+{
+    private bool $resolved = false;
+
+    public function __construct(private ?string $presentedPassword)
+    {
+    }
+
+    public function getPresentedPassword(): ?string
+    {
+        return $this->presentedPassword;
+    }
+
+    public function setResolved(): void
+    {
+        $this->resolved = true;
+    }
+
+    public function isResolved(): bool
+    {
+        return $this->resolved;
+    }
+}
```

Comment:
We might simplify the class as follows:
```suggestion
class PasswordStrengthBadge implements BadgeInterface
{

    public function __construct(private ?string $presentedPassword)
    {
    }

    public function getPresentedPassword(): ?string
    {
        return $this->presentedPassword;
    }

    public function isResolved(): bool
    {
        return false;
    }
}
```

Reply:
Not really. The badge must be resolved in order to pass authentication. See https://github.com/symfony/security-http/blob/5.4/Authentication/AuthenticatorManager.php#L190